src/cluster.c

```diff
@@ -1097,15 +1117,38 @@ getNodeByQuery(client *c, struct serverCommand *cmd, robj **argv, int argc, int
                 }
             }
 
+            /* Block MOVE command as the destination key is not expected to exist, and we don't know if it was migrated */
+            if ((migrating_slot || importing_slot) && mcmd->proc == moveCommand) {
+                if (error_code) *error_code = CLUSTER_REDIR_UNSTABLE;
+                getKeysFreeResult(&result);
+                return NULL;
+            }
+
+            /* Block the COPY command if it's cross-DB to keep the code simple.
+             * Allowing cross-DB COPY is possible, but it would require looking up the second key in the target DB.
+             * The command should only be allowed if the key exists. We may revisit this decision in the future. */
```

Comment:
Why not just block all cross-DB operations if we are going to block this one? I hadn't really considered that before this moment. The main ask was basically simple multi-tenancy, not doing material data manipulation across DBs. If we disallow selects in multi, move, rename (with a DB) and copy (with a DB), we're left with a smaller surface area for the getNodeByQuery. That removes a lot of this code off the hot path since we are not longer required to do a lot of validation to know if the command can be executed or not.

Reply:
The main benefit is to make it easy to migrate from standalone to cluster. If some operations are missing, then the obstacle still exists.

I agree we should optimize for the hot code path, but I think we should try to support the cross-DB operations too. Otherwise it's another subtle difference between standalone and cluster.

Reply:
> Regarding the code path - I am not sure why @madolson you think it is that complicated?

It's just code on the hot path of getNodeByQuery, which we should probably avoid as much as possible. Every check hurts branch prediction. It's not a lot but we should try to avoid it. 

None of this is a deal breaker for me though.